templates/etc/apt/sources.list.j2

```diff
@@ -27,7 +27,7 @@
 {%     endfor %}
 {%   else %}
 {%     for entry_uri in (entry.uri|d([]) + entry.uris|d([])) %}
-{%       if entry_uri in apt__tpl_seen_sources %}
+{%       if (entry_uri in apt__tpl_seen_sources and apt__sources_filter_duplicates|bool) %}
```

Comment:
Currently the template gathers all data from the `apt__*_sources` lists and compares the mirror URLs to list of seen mirrors. I suppose that comparsion could be more through by using dictionaries and checking for certain keys... Perhaps you think that the role scans the `/etc/apt/sources.list` file? Nah, that file doesn't exist anymore. :-)